internal/icinga2/api_responses.go

```diff
@@ -264,13 +263,52 @@ type DowntimeTriggered struct {
 //
 // NOTE:
 //   - An empty Service field indicates a host being in flapping state.
+//   - State includes the current state of the Checkable at the point in time at which it enters or exits the flapping state.
+//   - CurrentFlapping indicates the current flapping value of a Checkable in percent.
+//   - ThresholdLow is the low/min flapping threshold value set by the user (CurrentFlapping < ThresholdLow = flapping end).
+//   - ThresholdHigh is the high/max flapping threshold value set by the user (CurrentFlapping > ThresholdHigh = flapping start).
 //
 // https://icinga.com/docs/icinga-2/latest/doc/12-icinga2-api/#event-stream-type-flapping
 type Flapping struct {
-	Timestamp  UnixFloat `json:"timestamp"`
-	Host       string    `json:"host"`
-	Service    string    `json:"service"`
-	IsFlapping bool      `json:"is_flapping"`
+	Timestamp       UnixFloat `json:"timestamp"`
+	Host            string    `json:"host"`
+	Service         string    `json:"service"`
+	IsFlapping      bool      `json:"is_flapping"`
+	State           int       `json:"state"`
+	CurrentFlapping int       `json:"current_flapping"`
+	ThresholdLow    int       `json:"threshold_low"`
+	ThresholdHigh   int       `json:"threshold_high"`
+}
+
+// CheckableCreatedDeleted represents the Icinga 2 API stream Checkable created/deleted response.
+//
+// NOTE:
+//   - The ObjectName field may already contain the composed name of the checkable if the ObjectType is `Service`.
+//   - The EventType field indicates which event type is currently being streamed and is either
+//     set to typeObjectCreated or typeObjectDeleted.
+type CheckableCreatedDeleted struct {
+	ObjectName string `json:"object_name"`
+	ObjectType string `json:"object_type"`
+	EventType  string `json:"type"`
+}
+
+// GetHostName returns the host name of this current response result.
+func (c *CheckableCreatedDeleted) GetHostName() string {
+	if c.ObjectType == "Host" {
+		return c.ObjectName
+	}
+
+	return strings.Split(c.ObjectName, "!")[0]
+}
+
+// GetServiceName returns the service of this current response result.
+// Returns empty string if the current result type is of type `Host`.
+func (c *CheckableCreatedDeleted) GetServiceName() string {
+	if c.ObjectType == "Service" {
+		return strings.Split(c.ObjectName, "!")[1]
+	}
+
+	return ""
 }
```

Comment:
> This implementation relies on these events being filtered to hosts and services only.

There are only two types of Checkable objects, `Host` and `Service`, and that type is named after these, why would you want to use it with anything else?

Reply:
There are no other types of checkables, but other types of objects. This PR assumes that the `ObjectCreated` and `ObjectDeleted` event stream messages won't be used for anything but hosts and services:
 
https://github.com/Icinga/icinga-notifications/blob/ab7e482b635236aa147c4aa30a97cda1f8535988/internal/icinga2/api_responses.go#L362-L363

https://github.com/Icinga/icinga-notifications/blob/ab7e482b635236aa147c4aa30a97cda1f8535988/internal/icinga2/client_api.go#L352-L355

If the use of these event streams was extended to other types, this would result in strange behavior without closely checking all the code again (`GetHostName()` would assume it was a service, `GetServiceName()` would assume it was a host). On the other hand, if the event would just represent a generic ObjectCreated/Deleted event and the code consuming these events just checked if it's a host or service, this would be more robust.